txpool/slot_gauge.go

```diff
@@ -27,6 +31,12 @@ func (g *slotGauge) decrease(slots uint64) {
 	atomic.AddUint64(&g.height, ^(slots - 1))
 }
 
+// highPressure checks if the gauge level
+// is higher than the highPressureMark (0.8 * max)
+func (g *slotGauge) highPressure() bool {
+	return g.read() > uint64(highPressureMark*float64(g.max))
```

Comment:
There may be an error depends on `highPressureMark` and `g.max`. For example, the error of right term is 1 where highPressureMark=0.29 and max=100. I think it's ok as far as highPressureMark is 0.8. I recommend you to calculate by integer division as long as highPressureMark can be represented in percent without the value after decimal point.

Reply:
Can you write the suggestion as a code snippet ?

Reply:
```go
const (
     highPressureMark = 80 // 80%
)

func (g *slotGauge) highPressure() bool {
        return g.read() > (highPressureMark*g.max) / 100
}
```

Reply:
Great catch, I was not aware of [this behaviour](https://goplay.tools/snippet/RP60no2ZcyE) in Go.

(the `highPressureMark` is `const` anyhow, but maybe we'll make it configurable in the future)

[Refactored](https://github.com/0xPolygon/polygon-edge/pull/689/commits/e3eb589d38b2d4ff417f0a18db6cc50ff1e0bf48)